### ISSUE TITLE
Global variable overload?

### ISSUE THREAD

**Issue by JarvisTheJelly** [Pixdigit/Project-Interstellar#2]
I understand you are probably very comfortable with using 10s of globals in nearly every function in every file, but perhaps I could offer a different solution. Currently you have a lot of variables that are global that are used in a lot of different functions. Calling 'global <variable>' for every variable in every function you need it in can create messy and unnecessarily lengthy code. What I would suggest is creating a master 'World' class that stores all of those variables, and each function or class can store a copy or a reference to the instance of that class. That way you only have to call 'main_world.<variable>' and it is much easier to see global variables vs local variables.

**Comment by Pixdigit** [comment]
Nick,

I know that problem. But instead of using "only" a class I use a whole file
(settings.py) which is then called (and it has some extra functionallity).
But I might got your point wrong.
Am 16.07.2015 00:49 schrieb "Nick Jarvis" <notifications@github.com>:

**Comment by JarvisTheJelly** [comment]
My thought was just to make "settings" into a class. In settings.py alone there are 111 instances of the use of the word "global", taking up 109 lines, which could all be removed. I'll make a simple model of what I mean and send it to you.

**Comment by Pixdigit** [comment]
Ok.
But another thing to consider is that this way we have a lookup table of what which variable means.

**Comment by JarvisTheJelly** [comment]
Nothing would really change, except all of the 'global' declarations would be removed and 'self.' would be inserted in. I'll send that sample file in a few minutes.

**Comment by Pixdigit** [comment]
I see what you mean and because you have more experience I think we could (after fixing everything) merge the remove-globals branch. But I think we should at least keep a separate file in which all settings-variables are shortly explained what they are used for. Which is really helpful if one works on a part one only barely understands. Explanation of the variables is really helpful. Especially when there name is confusing or irritating.

**Comment by JarvisTheJelly** [comment]
Yes, either another file or just add comments to blocks of related variables like, "These boolean values are for what buttons the player is pressing on the current frame"

**Comment by Pixdigit** [comment]
Well two years later and yet the branch got not merged. But since I am doing a partial rewrite I will implement your idea on the go.
